tests/fixtures/markup_rendering.yml

```diff
@@ -17,9 +17,47 @@ textarea:
     '#title': Textarea
     '#value': Text
   assert:
-    contains:
-      - 'Textarea'
-      - 'Text'
+    equals:
+      'label.form-label': "Textarea"
+      'textarea.form-control': "Text"
     count:
       'textarea.form-textarea': 1
       'textarea.form-control': 1
+checkbox_assertion:
+  render:
+    '#type': checkbox
+    '#title': Checkbox
+    '#attributes':
+      'foo': bar
+  assert:
+    contains:
+      - 'Checkbox'
+      - 'foo="bar"'
+    count:
+      'label.form-check-label': 1
+      'input.form-checkbox.form-check-input': 1
+select:
+  render:
+    '#type': select
+    '#title': Select
+    '#options':
+      - One
+      - Two
+      - Three
+  assert:
+    equals:
+      'label.form-label': "Select"
+      'option[value=0]': "One"
+      'option[value=1]': "Two"
+      'option[value=2]': "Three"
+    count:
+      'label.form-label': 1
+      'select.form-select': 1
+submit:
```

Comment:
lets do more variants for all these, example here in submit we can:
check for btn-primary, btn-secondary, and a disabled one